Using GPRT as a subproject

I've been using GPRT as a subproject in https://github.com/General-Purpose-Raytracing-Toolkit/h5m-reader and am running into some issues compiling an executable.

First, I needed to include the CMake setting

`set(CMAKE_POSITION_INDEPENDENT_CODE ON)`

to allow symbols to be relocated.

The strange part is that I can successfully compile only if I first build GPRT as a shared library _then_ as a static library (or vice versa). Building one vs the other first results in a different set of undefined symbol errors (provided below)


### Building Shared First

Linker Lines:

```
usr/bin/cmake -E cmake_link_script CMakeFiles/gprt-h5m.dir/link.txt --verbose=1
/usr/bin/c++ CMakeFiles/gprt-h5m.dir/src/main.cpp.o -o gprt-h5m  -Wl,-rpath,/home/pshriwise/opt/lib:/home/pshriwise/soft/h5m-reader/bld/3rdParty/gprt/gprt:/usr/lib/x86_64-linux-gnu/hdf5/serial /home/pshriwise/opt/lib/libMOAB.so.5.4.0 3rdParty/gprt/gprt/libgprt.so /usr/lib/x86_64-linux-gnu/libnetcdf.so /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5_hl.so /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so /usr/lib/x86_64-linux-gnu/libcrypto.so /usr/lib/x86_64-linux-gnu/libcurl.so /usr/lib/x86_64-linux-gnu/libpthread.a /usr/lib/x86_64-linux-gnu/libsz.so /usr/lib/x86_64-linux-gnu/libz.so /usr/lib/x86_64-linux-gnu/libdl.a -lm /usr/lib/x86_64-linux-gnu/libcurl.so /usr/lib/x86_64-linux-gnu/libpthread.a /usr/lib/x86_64-linux-gnu/libsz.so /usr/lib/x86_64-linux-gnu/libz.so /usr/lib/x86_64-linux-gnu/libdl.a -lm 
```

Symbol Errors:
```
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetPhysicalDeviceFeatures'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkAllocateCommandBuffers'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkBindBufferMemory'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkFreeCommandBuffers'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkBeginCommandBuffer'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkMapMemory'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkDestroyDevice'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCreatePipelineLayout'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCmdCopyBuffer'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCreateComputePipelines'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkDestroyInstance'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkEndCommandBuffer'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetPhysicalDeviceProperties2'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCreateShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCreateCommandPool'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCmdDispatch'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetPhysicalDeviceQueueFamilyProperties'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetPhysicalDeviceProperties'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkFreeMemory'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetBufferMemoryRequirements'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkAllocateMemory'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCmdPushConstants'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCreateInstance'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkEnumerateInstanceLayerProperties'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkDestroyBuffer'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkUnmapMemory'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCreateBuffer'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetDeviceQueue'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkQueueSubmit'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCreateDevice'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetPhysicalDeviceFeatures2'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkDestroyCommandPool'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkDestroyShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkDestroyPipelineLayout'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkDestroyPipeline'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkInvalidateMappedMemoryRanges'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkEnumerateInstanceExtensionProperties'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkCmdBindPipeline'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetPhysicalDeviceMemoryProperties'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkEnumerateDeviceExtensionProperties'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetInstanceProcAddr'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkGetDeviceProcAddr'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkEnumeratePhysicalDevices'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.so: undefined reference to `vkFlushMappedMemoryRanges'
collect2: error: ld returned 1 exit status
```

### Building Static First

Linker Line:

```
/usr/bin/cmake -E cmake_link_script CMakeFiles/gprt-h5m.dir/link.txt --verbose=1
/usr/bin/c++ CMakeFiles/gprt-h5m.dir/src/main.cpp.o -o gprt-h5m  -Wl,-rpath,/home/pshriwise/opt/lib:/usr/lib/x86_64-linux-gnu/hdf5/serial /home/pshriwise/opt/lib/libMOAB.so.5.4.0 3rdParty/gprt/gprt/libgprt.a /usr/lib/x86_64-linux-gnu/libnetcdf.so /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5_hl.so /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so /usr/lib/x86_64-linux-gnu/libcrypto.so /usr/lib/x86_64-linux-gnu/libcurl.so /usr/lib/x86_64-linux-gnu/libpthread.a /usr/lib/x86_64-linux-gnu/libsz.so /usr/lib/x86_64-linux-gnu/libz.so /usr/lib/x86_64-linux-gnu/libdl.a -lm /usr/lib/x86_64-linux-gnu/libcurl.so /usr/lib/x86_64-linux-gnu/libpthread.a /usr/lib/x86_64-linux-gnu/libsz.so /usr/lib/x86_64-linux-gnu/libz.so /usr/lib/x86_64-linux-gnu/libdl.a -lm 
```

Symbol Errors:

```
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `gprtRayGenLaunch3D':
gprt.cpp:(.text+0x3032): undefined reference to `vkBeginCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text+0x3061): undefined reference to `vkCmdBindPipeline'
/usr/bin/ld: gprt.cpp:(.text+0x30b6): undefined reference to `vkCmdPushConstants'
/usr/bin/ld: gprt.cpp:(.text+0x3495): undefined reference to `vkEndCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text+0x3597): undefined reference to `vkQueueSubmit'
/usr/bin/ld: gprt.cpp:(.text+0x3620): undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `gprtComputeLaunch3D':
gprt.cpp:(.text+0x3950): undefined reference to `vkBeginCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text+0x397f): undefined reference to `vkCmdBindPipeline'
/usr/bin/ld: gprt.cpp:(.text+0x39d4): undefined reference to `vkCmdPushConstants'
/usr/bin/ld: gprt.cpp:(.text+0x3c2b): undefined reference to `vkEndCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text+0x3d2d): undefined reference to `vkQueueSubmit'
/usr/bin/ld: gprt.cpp:(.text+0x3db6): undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Buffer::map(unsigned long, unsigned long)':
gprt.cpp:(.text._ZN6Buffer3mapEmm[_ZN6Buffer3mapEmm]+0xab): undefined reference to `vkMapMemory'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer3mapEmm[_ZN6Buffer3mapEmm]+0x106): undefined reference to `vkBeginCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer3mapEmm[_ZN6Buffer3mapEmm]+0x1d3): undefined reference to `vkCmdCopyBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer3mapEmm[_ZN6Buffer3mapEmm]+0x1e9): undefined reference to `vkEndCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer3mapEmm[_ZN6Buffer3mapEmm]+0x2d0): undefined reference to `vkQueueSubmit'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer3mapEmm[_ZN6Buffer3mapEmm]+0x347): undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer3mapEmm[_ZN6Buffer3mapEmm]+0x407): undefined reference to `vkMapMemory'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Buffer::unmap()':
gprt.cpp:(.text._ZN6Buffer5unmapEv[_ZN6Buffer5unmapEv]+0x6e): undefined reference to `vkUnmapMemory'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer5unmapEv[_ZN6Buffer5unmapEv]+0xdb): undefined reference to `vkBeginCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer5unmapEv[_ZN6Buffer5unmapEv]+0x1a8): undefined reference to `vkCmdCopyBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer5unmapEv[_ZN6Buffer5unmapEv]+0x1be): undefined reference to `vkEndCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer5unmapEv[_ZN6Buffer5unmapEv]+0x2a5): undefined reference to `vkQueueSubmit'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer5unmapEv[_ZN6Buffer5unmapEv]+0x31c): undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer5unmapEv[_ZN6Buffer5unmapEv]+0x3b7): undefined reference to `vkUnmapMemory'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Buffer::flush(unsigned long, unsigned long)':
gprt.cpp:(.text._ZN6Buffer5flushEmm[_ZN6Buffer5flushEmm]+0x89): undefined reference to `vkFlushMappedMemoryRanges'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Buffer::invalidate(unsigned long, unsigned long)':
gprt.cpp:(.text._ZN6Buffer10invalidateEmm[_ZN6Buffer10invalidateEmm]+0x89): undefined reference to `vkInvalidateMappedMemoryRanges'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Buffer::destroy()':
gprt.cpp:(.text._ZN6Buffer7destroyEv[_ZN6Buffer7destroyEv]+0x3e): undefined reference to `vkDestroyBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer7destroyEv[_ZN6Buffer7destroyEv]+0x7f): undefined reference to `vkFreeMemory'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer7destroyEv[_ZN6Buffer7destroyEv]+0xc0): undefined reference to `vkDestroyBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6Buffer7destroyEv[_ZN6Buffer7destroyEv]+0x101): undefined reference to `vkFreeMemory'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Buffer::Buffer(VkPhysicalDevice_T*, VkDevice_T*, VkCommandBuffer_T*, VkQueue_T*, unsigned int, unsigned int, unsigned long, void*)':
gprt.cpp:(.text._ZN6BufferC2EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv[_ZN6BufferC5EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv]+0x192): undefined reference to `vkGetPhysicalDeviceMemoryProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN6BufferC2EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv[_ZN6BufferC5EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv]+0x24b): undefined reference to `vkCreateBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6BufferC2EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv[_ZN6BufferC5EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv]+0x3f4): undefined reference to `vkCreateBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN6BufferC2EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv[_ZN6BufferC5EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv]+0x541): undefined reference to `vkGetBufferMemoryRequirements'
/usr/bin/ld: gprt.cpp:(.text._ZN6BufferC2EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv[_ZN6BufferC5EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv]+0x601): undefined reference to `vkAllocateMemory'
/usr/bin/ld: gprt.cpp:(.text._ZN6BufferC2EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv[_ZN6BufferC5EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv]+0x739): undefined reference to `vkBindBufferMemory'
/usr/bin/ld: gprt.cpp:(.text._ZN6BufferC2EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv[_ZN6BufferC5EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv]+0x846): undefined reference to `vkGetBufferMemoryRequirements'
/usr/bin/ld: gprt.cpp:(.text._ZN6BufferC2EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv[_ZN6BufferC5EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv]+0x8fe): undefined reference to `vkAllocateMemory'
/usr/bin/ld: gprt.cpp:(.text._ZN6BufferC2EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv[_ZN6BufferC5EP18VkPhysicalDevice_TP10VkDevice_TP17VkCommandBuffer_TP9VkQueue_TjjmPv]+0xa36): undefined reference to `vkBindBufferMemory'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Compute::Compute(VkDevice_T*, Module*, char const*, unsigned long, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, _GPRTVarDef, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, _GPRTVarDef> > >)':
gprt.cpp:(.text._ZN7ComputeC2EP10VkDevice_TP6ModulePKcmSt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE11_GPRTVarDefSt4hashISC_ESt8equal_toISC_ESaISt4pairIKSC_SD_EEE[_ZN7ComputeC5EP10VkDevice_TP6ModulePKcmSt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE11_GPRTVarDefSt4hashISC_ESt8equal_toISC_ESaISt4pairIKSC_SD_EEE]+0x30c): undefined reference to `vkCreateShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Compute::destroy()':
gprt.cpp:(.text._ZN7Compute7destroyEv[_ZN7Compute7destroyEv]+0x5a): undefined reference to `vkDestroyShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `RayGen::RayGen(VkDevice_T*, Module*, char const*, unsigned long, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, _GPRTVarDef, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, _GPRTVarDef> > >)':
gprt.cpp:(.text._ZN6RayGenC2EP10VkDevice_TP6ModulePKcmSt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE11_GPRTVarDefSt4hashISC_ESt8equal_toISC_ESaISt4pairIKSC_SD_EEE[_ZN6RayGenC5EP10VkDevice_TP6ModulePKcmSt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE11_GPRTVarDefSt4hashISC_ESt8equal_toISC_ESaISt4pairIKSC_SD_EEE]+0x30e): undefined reference to `vkCreateShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `RayGen::destroy()':
gprt.cpp:(.text._ZN6RayGen7destroyEv[_ZN6RayGen7destroyEv]+0x5a): undefined reference to `vkDestroyShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Miss::Miss(VkDevice_T*, Module*, char const*, unsigned long, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, _GPRTVarDef, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, _GPRTVarDef> > >)':
gprt.cpp:(.text._ZN4MissC2EP10VkDevice_TP6ModulePKcmSt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE11_GPRTVarDefSt4hashISC_ESt8equal_toISC_ESaISt4pairIKSC_SD_EEE[_ZN4MissC5EP10VkDevice_TP6ModulePKcmSt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE11_GPRTVarDefSt4hashISC_ESt8equal_toISC_ESaISt4pairIKSC_SD_EEE]+0x30c): undefined reference to `vkCreateShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Miss::destroy()':
gprt.cpp:(.text._ZN4Miss7destroyEv[_ZN4Miss7destroyEv]+0x5a): undefined reference to `vkDestroyShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `GeomType::setClosestHit(int, Module*, char const*)':
gprt.cpp:(.text._ZN8GeomType13setClosestHitEiP6ModulePKc[_ZN8GeomType13setClosestHitEiP6ModulePKc]+0x228): undefined reference to `vkCreateShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `GeomType::setAnyHit(int, Module*, char const*)':
gprt.cpp:(.text._ZN8GeomType9setAnyHitEiP6ModulePKc[_ZN8GeomType9setAnyHitEiP6ModulePKc]+0x228): undefined reference to `vkCreateShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `GeomType::setIntersection(int, Module*, char const*)':
gprt.cpp:(.text._ZN8GeomType15setIntersectionEiP6ModulePKc[_ZN8GeomType15setIntersectionEiP6ModulePKc]+0x228): undefined reference to `vkCreateShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `GeomType::destroy()':
gprt.cpp:(.text._ZN8GeomType7destroyEv[_ZN8GeomType7destroyEv]+0x97): undefined reference to `vkDestroyShaderModule'
/usr/bin/ld: gprt.cpp:(.text._ZN8GeomType7destroyEv[_ZN8GeomType7destroyEv]+0x117): undefined reference to `vkDestroyShaderModule'
/usr/bin/ld: gprt.cpp:(.text._ZN8GeomType7destroyEv[_ZN8GeomType7destroyEv]+0x197): undefined reference to `vkDestroyShaderModule'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `TriangleAccel::build(std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, Stage, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, Stage> > >)':
gprt.cpp:(.text._ZN13TriangleAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13TriangleAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x899): undefined reference to `vkBeginCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN13TriangleAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13TriangleAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x945): undefined reference to `vkEndCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN13TriangleAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13TriangleAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0xa30): undefined reference to `vkQueueSubmit'
/usr/bin/ld: gprt.cpp:(.text._ZN13TriangleAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13TriangleAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0xaa4): undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `AABBAccel::build(std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, Stage, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, Stage> > >)':
gprt.cpp:(.text._ZN9AABBAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN9AABBAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x90d): undefined reference to `vkBeginCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN9AABBAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN9AABBAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x9b9): undefined reference to `vkEndCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN9AABBAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN9AABBAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0xaa4): undefined reference to `vkQueueSubmit'
/usr/bin/ld: gprt.cpp:(.text._ZN9AABBAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN9AABBAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0xb18): undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `InstanceAccel::build(std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, Stage, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, Stage> > >)':
gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x1f8): undefined reference to `vkBeginCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x277): undefined reference to `vkCmdPushConstants'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x2f9): undefined reference to `vkCmdBindPipeline'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x349): undefined reference to `vkCmdDispatch'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x35c): undefined reference to `vkEndCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x3ec): undefined reference to `vkQueueSubmit'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0x472): undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0xb4f): undefined reference to `vkBeginCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0xc0d): undefined reference to `vkEndCommandBuffer'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0xcf9): undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0xd23): undefined reference to `vkQueueSubmit'
/usr/bin/ld: gprt.cpp:(.text._ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE[_ZN13InstanceAccel5buildESt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE5StageSt4lessIS6_ESaISt4pairIKS6_S7_EEE]+0xda9): undefined reference to `vkQueueWaitIdle'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Context::Context(int*, int)::{lambda(unsigned int, unsigned int)#4}::operator()(unsigned int, unsigned int) const':
gprt.cpp:(.text._ZZN7ContextC4EPiiENKUljjE2_clEjj[_ZZN7ContextC4EPiiENKUljjE2_clEjj]+0x73): undefined reference to `vkCreateCommandPool'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Context::Context(int*, int)':
gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x317): undefined reference to `vkEnumerateInstanceExtensionProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x38a): undefined reference to `vkEnumerateInstanceExtensionProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x7ea): undefined reference to `vkEnumerateInstanceLayerProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x84d): undefined reference to `vkEnumerateInstanceLayerProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x976): undefined reference to `vkCreateInstance'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0xa43): undefined reference to `vkGetInstanceProcAddr'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0xa6a): undefined reference to `vkGetInstanceProcAddr'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0xb76): undefined reference to `vkEnumeratePhysicalDevices'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0xd7b): undefined reference to `vkEnumeratePhysicalDevices'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0xe92): undefined reference to `vkGetPhysicalDeviceProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x10cf): undefined reference to `vkGetPhysicalDeviceProperties2'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x116c): undefined reference to `vkGetPhysicalDeviceFeatures2'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x11cc): undefined reference to `vkGetPhysicalDeviceProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1278): undefined reference to `vkGetPhysicalDeviceProperties2'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x129c): undefined reference to `vkGetPhysicalDeviceFeatures'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x12c0): undefined reference to `vkGetPhysicalDeviceMemoryProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x12e2): undefined reference to `vkGetPhysicalDeviceQueueFamilyProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1369): undefined reference to `vkGetPhysicalDeviceQueueFamilyProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1397): undefined reference to `vkEnumerateDeviceExtensionProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1415): undefined reference to `vkEnumerateDeviceExtensionProperties'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1b9d): undefined reference to `vkGetPhysicalDeviceFeatures2'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1e4e): undefined reference to `vkCreateDevice'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1f11): undefined reference to `vkGetDeviceProcAddr'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1f3b): undefined reference to `vkGetDeviceProcAddr'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1f65): undefined reference to `vkGetDeviceProcAddr'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1f8f): undefined reference to `vkGetDeviceProcAddr'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1fb9): undefined reference to `vkGetDeviceProcAddr'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o):gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x1fe3): more undefined references to `vkGetDeviceProcAddr' follow
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Context::Context(int*, int)':
gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x21c3): undefined reference to `vkAllocateCommandBuffers'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x22a9): undefined reference to `vkAllocateCommandBuffers'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x238f): undefined reference to `vkAllocateCommandBuffers'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x246b): undefined reference to `vkGetDeviceQueue'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x24a4): undefined reference to `vkGetDeviceQueue'
/usr/bin/ld: gprt.cpp:(.text._ZN7ContextC2EPii[_ZN7ContextC5EPii]+0x24dd): undefined reference to `vkGetDeviceQueue'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Context::destroy()':
gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x42): undefined reference to `vkDestroyPipelineLayout'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x78): undefined reference to `vkDestroyPipeline'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0xae): undefined reference to `vkDestroyPipelineLayout'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0xe4): undefined reference to `vkDestroyPipeline'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x11a): undefined reference to `vkDestroyShaderModule'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x15d): undefined reference to `vkFreeCommandBuffers'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x18e): undefined reference to `vkFreeCommandBuffers'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x1bf): undefined reference to `vkFreeCommandBuffers'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x1e5): undefined reference to `vkDestroyCommandPool'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x20b): undefined reference to `vkDestroyCommandPool'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x231): undefined reference to `vkDestroyCommandPool'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x249): undefined reference to `vkDestroyDevice'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context7destroyEv[_ZN7Context7destroyEv]+0x275): undefined reference to `vkDestroyInstance'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Context::buildPrograms()':
gprt.cpp:(.text._ZN7Context13buildProgramsEv[_ZN7Context13buildProgramsEv]+0x164): undefined reference to `vkCreatePipelineLayout'
/usr/bin/ld: 3rdParty/gprt/gprt/libgprt.a(gprt.cpp.o): in function `Context::setupInternalStages(Module*)':
gprt.cpp:(.text._ZN7Context19setupInternalStagesEP6Module[_ZN7Context19setupInternalStagesEP6Module]+0x132): undefined reference to `vkCreatePipelineLayout'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context19setupInternalStagesEP6Module[_ZN7Context19setupInternalStagesEP6Module]+0x2c7): undefined reference to `vkCreateShaderModule'
/usr/bin/ld: gprt.cpp:(.text._ZN7Context19setupInternalStagesEP6Module[_ZN7Context19setupInternalStagesEP6Module]+0x422): undefined reference to `vkCreateComputePipelines'
collect2: error: ld returned 1 exit status
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -45,14 +45,23 @@ jobs:
       run: |
         ./util/download-dxc.sh
 
-    - name: Build
+    - name: Build Static
       shell: bash
       run: |
-        mkdir bld
-        cd bld
+        mkdir bld-static
+        cd bld-static
         cmake --version
         cmake .. -DVulkan_BIN_DIR=$HOME/dxc-artifacts/bin
-        cmake .. -DVulkan_BIN_DIR=$HOME/dxc-artifacts/bin
+        make VERBOSE=1 -j4
+
+    - name: Build Shared
+      shell: bash
+      run: |
+        mkdir bld-shared
+        cd bld-shared
+        cmake --version
+        cmake .. -DVulkan_BIN_DIR=$HOME/dxc-artifacts/bin \
+                 -DGPRT_BUILD_SHARED=ON
         make VERBOSE=1 -j4
 
     # Just checking build until a CPU backend is ready

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,14 @@ if (NOT GPRT_IS_SUBPROJECT)
 endif()
 
 # ------------------------------------------------------------------
+# Configuration Options
+# ------------------------------------------------------------------
+option(GPRT_BUILD_SHARED "Build GPRT as a shared library? (otherwise static)" OFF)
+set(BUILD_SHARED_LIBS ${GPRT_BUILD_SHARED}) # use 'GPRT_' naming convention
+
+option(GPRT_BUILD_SAMPLES "Build the Samples?" ON)
+
+# ------------------------------------------------------------------
 # first, include some dependencies
 # ------------------------------------------------------------------
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/gprt/cmake/")
@@ -53,24 +61,22 @@ endif()
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+if (GPRT_BUILD_SHARED)
+  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+endif()
 
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 
 option(OWL_BUILD_SAMPLES "Build the Samples?" ON)
 
+add_subdirectory(3rdParty)
+
+
 # ------------------------------------------------------------------
 # gprt library itself
 # ------------------------------------------------------------------
-option(GPRT_BUILD_SHARED "Build GPRT as a shared library? (otherwise static)" OFF)
-set(BUILD_SHARED_LIBS ${GPRT_BUILD_SHARED}) # use 'GPRT_' naming convention
 add_subdirectory(gprt)
-
-option(GPRT_BUILD_SAMPLES "Build the Samples?" ON)
-# add_compile_definitions(GPRT_BUILDING_ALL_SAMPLES)
-# option(GPRT_BUILD_ADVANCED_TESTS "Build the *advanced* test-cases?" OFF)
-
-add_subdirectory(3rdParty)
 
 # ------------------------------------------------------------------
 # tutorial/samples

--- a/gprt/CMakeLists.txt
+++ b/gprt/CMakeLists.txt
@@ -21,7 +21,9 @@
 # SOFTWARE.
 
 add_library(gprt)
-
+if (GPRT_BUILD_SHARED)
+  set_property(TARGET gprt PROPERTY POSITION_INDEPENDENT_CODE ON)
+endif()
 include(cmake/hlsl_embed_spirv.cmake)
 
 embed_spirv(
@@ -58,9 +60,9 @@ set_target_properties(gprt
 #     )
 # ENDIF(WIN32)
 
-target_link_libraries(gprt 
-    PUBLIC 
-    ${Vulkan_LIBRARY} 
+target_link_libraries(gprt
+    PUBLIC
+    ${Vulkan_LIBRARY}
     gprtDeviceCode
 )
 

